ert3/engine/_run.py

```diff
@@ -10,22 +9,31 @@
 _SOURCE_SEPARATOR = "."
 
 
+# pylint: disable=too-many-arguments
 def _prepare_experiment(
     workspace_root: pathlib.Path,
     experiment_name: str,
     ensemble: ert3.config.EnsembleConfig,
+    stages_config: ert3.config.StagesConfig,
     ensemble_size: int,
     parameters_config: ert3.config.ParametersConfig,
 ) -> None:
     if ert3.workspace.experiment_has_run(workspace_root, experiment_name):
         raise ValueError(f"Experiment {experiment_name} have been carried out.")
 
+    step = stages_config.step_from_key(ensemble.forward_model.stage)
+    if not step:
+        raise ValueError(f"No step for key {ensemble.forward_model.stage}")
+
     parameters: Dict[str, List[str]] = {}
     for input_record in ensemble.input:
         record_name = input_record.record
         record_source = input_record.source.split(_SOURCE_SEPARATOR, maxsplit=1)
+        record_mime = next(
+            input_.mime for input_ in step.input if input_.record == record_name
+        )
```

Comment:
Let's make an issue for this as well, so we don't forget to fix this :slightly_smiling_face:

Reply:
https://github.com/equinor/ert/issues/1965